akka-stream/src/main/scala/akka/stream/RestartSettings.scala

```diff
@@ -76,7 +87,8 @@ object RestartSettings {
       randomFactor = randomFactor,
       maxRestarts = Int.MaxValue,
       maxRestartsWithin = minBackoff,
-      logSettings = LogSettings.defaultSettings)
+      logSettings = LogSettings.defaultSettings,
+      restartOn = _ => true)
```

Comment:
Re-use the already defined `akka.util.ConstantFun.anyToTrue` as default for these

Reply:
Is it worth adding a `javaAlwaysTruePredicate`?

Reply:
Yeah, please go ahead